### DISCUSSION TITLE
About GitHub’s CopilotForXcode

### DISCUSSION THREAD

**Discussion by intitni** [intitni/CopilotForXcode#597] [Announcements]
Yes, we had a deal so it’s 100% legitimate. 

This project will still be active, though.

**Comment by SinOverCos** [comment]
How will the two active forks be maintained? Will they pull from each other once in a while? How should one decide whether to use intitni/CopilotForXcode or github/CopilotForXcode?

**Reply by intitni** [reply]
They are different apps that provide different features, and have different goals and priorities. Just choose the one the suits your needs.

This repo will not pull from other repos.

**Comment by ladislas** [comment]
is there a comparison available between the two somewhere?
i've given the Github's version a try today and it feels like it doesn't work as well as yours.

but I'm curious what the similarities and differences are.

**Reply by intitni** [reply]
Apparently, GitHub's version focuses more on the GitHub Copilot features, and their goal is to align the extension to the VSCode extension.

On the other hand, GitHub Copilot is only one of the service providers in this repo, the original Copilot for Xcode. You can use different services like Codeium and other LLMs, too. We support many other features that are more Xcode-oriented and not likely to be implemented in GitHub's version, such as `automatically closing idle tabs in Xcode`. 

We are not aiming to align the extension to the GitHub Copilot VSCode extension, so we are handling code completions, inline suggestions, or chat in different ways, and they will be more different in the future.

**Reply by ladislas** [reply]
thanks a lot! :)

**Comment by TroutZhang** [comment]
Cool to find here!

I think this also explains why I am seeing from GitHub's version today, that I met https://github.com/intitni/CopilotForXcode/issues/429 again.

Basically, I picked some random Cpp files to test. Most work, but 1 doesn't. The 1 that doesn't will be the same as the 429 bug that pressing tab makes you go to the file head.
* this file is 4000 lines in length, maybe it could be the reason; since other I test OK with are no more than a few hundred lines.

I didn't fire a bug because now it seems OKish.
* That OKish is when it reaches the file head like the 429 bug; instead of staying there, now it will quickly jump back to where I was. It didn't jump back 2 hours ago, that's why I am searching the web and got here.
* so it looks like a flash now.
* it shows the same on my two MBPs, one m1max, one m3max, both with latest macOS 15.1 and Xcode 16.1
* wifi did switch from high speed office to middle speed home, though I don't know why high speed shows the bug.
* when it happens, I did see that yellow popup saying some "using fallback, need restart Xcode", it disappears too fast to catch the whole message.

**Reply by intitni** [reply]
It's not a bug but a limitation of an Xcode extension, it has nothing to do with file length. Please check the [FAQ for details](https://copilotforxcode.intii.com/faq/app-faq/#q-the-editor-jumps-when-i-accept-a-suggestion).

If you find a bug in GitHub's version, please consider submitting an issue in their repo since I am not sure how much orignal code are kept or modified.

**Reply by TroutZhang** [reply]
Thank you for creating such a good tool! I am from VS World, and this is the first Xcode extension I have installed!
